src/components/test-slot/candidate-link/candidate-link.ts

```diff
@@ -30,15 +37,85 @@ export class CandidateLinkComponent {
   @Input()
   testComplete: boolean;
 
-  @Input()
   isPortrait: boolean;
 
+  truncatedName: string;
+
   constructor(
+    private cd: ChangeDetectorRef,
+    private ellipses: EllipsisPipe,
     public modalController: ModalController,
     public app: AppComponent,
   ) {
   }
 
+  async ngOnInit() {
+    this.truncatedName = this.getName();
+    await this.monitorOrientation();
+  }
+
+  truncateValue() {
+    let finalSum = 0;
+
+    switch (this.app.getTextZoomClass()) {
+      case 'text-zoom-regular':
+        finalSum += 35;
+        if (!this.isPortrait) {
+          finalSum += 13;
+        }
+        break;
+      case 'text-zoom-large':
+        finalSum += 29;
+        if (!this.isPortrait) {
+          finalSum += 10;
+        }
+        break;
+      case 'text-zoom-x-large':
+        finalSum += 19;
+        if (!this.isPortrait) {
+          finalSum += 10;
+        }
+        break;
+      default:
+        finalSum += 100;
+        if (!this.isPortrait) {
+          finalSum += 30;
+        }
+        break;
+    }
+
+    finalSum -= (this.slot.booking.application.testCategory.length);
```

Comment:
We need to add some guards to accessing this data, for non test slots we wont have an `application` object